engine/src/main/java/org/camunda/bpm/engine/impl/pvm/runtime/operation/PvmAtomicOperationDeleteCascadeFireActivityEnd.java

```diff
@@ -71,6 +72,9 @@ protected void eventNotificationsCompleted(PvmExecutionImpl execution) {
 
     } else {
       if (execution.isScope()) {
+        if(execution instanceof ExecutionEntity && !execution.isProcessInstanceExecution() && execution.isCanceled()) {
+          execution.setSkipIoMappings(execution.isSkipIoMappings() || execution.getProcessEngine().getProcessEngineConfiguration().isSkipOutputMappingOnCanceledTasks());
+        }
```

Comment:
It took quite a while for me to understand which elements we are trying to get to here specifically. 
What do you think about adding a little one-line explanation here to state which executions this is meant for?

Also, how confident are we that we do not include any unwanted executions with this as well? I have to admit that getting a good grip on which executions this might unwantedly include is quite tough. From a gut feeling, I would say this could also include other activity types, e.g. user tasks with boundary events?

Do you think it makes sense to add some test cases for other activity types to ensure they are not affected by this? Or, if we already know there will be others affected by this, add test cases for them as well of course?

Reply:
As discussed I simplified the condition and made sure only external tasks are included. Will discuss this decision next year. Let's put this on hold for now.